### PR TITLE
Use rascaline calculators inside metatensor's atomistic models

### PR DESCRIPTION
**Pull request by Luthaf**
This PR does two things:

- replace `rascaline.torch.System` with `metatensor.torch.atomistic.System`
- implement `requested_neighbors_lists` to tell the simulation engine which neighbors lists we need for rascaline

This builds on top of https://github.com/lab-cosmo/metatensor/pull/405

----
:books: Documentation preview :books:: https://rascaline--253.org.readthedocs.build/en/253/

### CODE REVIEW

File: docs/src/references/api/torch/systems.rst [hunk 1]
```diff
@@ -1,7 +1,4 @@
 System
 ======
 
-.. autoclass:: rascaline.torch.System
```
Comment (PicoCentauri) [inline docs/src/references/api/torch/systems.rst:4]: Maybe add a comment that we are using the metatensor systems class and we provide some conversion functions.

**Review by PicoCentauri — COMMENTED** [review]
A quick comment. I do a deeper review asap.

**Review by PicoCentauri — APPROVED** [review]
Besides my comment to the docs it looks very good.

File: docs/src/references/api/torch/systems.rst (hunk 1 above)
Comment (Luthaf) [reply docs/src/references/api/torch/systems.rst:4]: done!

**Review by PicoCentauri — APPROVED** [review]

**Pull request merged**